### PR TITLE
Dw 5598 switch to rbac2 for development

### PR DESCRIPTION
**Pull request by caldwelltom**
toggled development back to rbac version 1

### CODE REVIEW

File: ci/aws-analytical-env/shared/meta.yml [hunk 1]
```diff
@@ -50,6 +50,9 @@ meta:
           args:
             - -exc
             - |
+              apk add -q --update --no-cache python3 && ln -sf python3 /usr/bin/python
```
Comment (IanUtley) [inline ci/aws-analytical-env/shared/meta.yml:53]: Why are you installing python? None of the things we are executing use python

File: ci/aws-analytical-env/shared/meta.yml [hunk 2]
```diff
@@ -71,6 +74,9 @@ meta:
...
```
Comment (IanUtley) [inline ci/aws-analytical-env/shared/meta.yml:77]: Again why are you installing python?

File: terraform/deploy/app/modules.tf [hunk 3]
```diff
@@ -2,7 +2,7 @@ module "emr_ami" {
   source = "../../modules/amis"
 
   providers = {
-    aws = aws.management
+    aws = aws.management-ami
```
Comment (IanUtley) [inline terraform/deploy/app/modules.tf:5]: Why was this changed? What problem did this solve?

File: ci/aws-analytical-env/shared/meta.yml (hunk 1 above)
Comment (caldwelltom) [reply ci/aws-analytical-env/shared/meta.yml:53]: @IanUtley - I think it is for the "data_cognito_user_roles.py" script?

Comment (IanUtley) [reply ci/aws-analytical-env/shared/meta.yml:53]: Ok, its because we are using external datasource.

File: terraform/deploy/app/modules.tf (hunk 3 above)
Comment (caldwelltom) [reply terraform/deploy/app/modules.tf:5]: @IanUtley This was changed during the merge - had been added more recently so thought it was required. Can change back?

File: terraform/modules/data_user_roles/data_cognito_user_roles.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    tf_input = json.loads(sys.stdin.read())
+
+    for var in ['role', 'user_pool_id', 'account_id', 'region']:
+        if var not in tf_input:
+            sys.stderr.write("Missing required variable {}".format(var))
+            sys.exit(1)
+
+    session = get_session(tf_input['role'], tf_input['region'])
+
+    users = get_cognito_users(session.client('cognito-idp'), tf_input['user_pool_id'])
```
Comment (IanUtley) [inline terraform/modules/data_user_roles/data_cognito_user_roles.py:44]: You aren't filtering users to those who are active.

File: terraform/modules/data_user_roles/data_cognito_user_roles.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        aws_secret_access_key=creds['SecretAccessKey'],
+        aws_session_token=creds['SessionToken'],
+        region_name=region
+    )
+
+
+def get_cognito_users(cognitoidp_client, user_pool_id):
+    res = cognitoidp_client.list_users(
+        UserPoolId=user_pool_id,
+        Filter="cognito:user_status=\"CONFIRMED\""
```
Comment (IanUtley) [inline terraform/modules/data_user_roles/data_cognito_user_roles.py:25]: This filter excludes ADFS users

File: terraform/modules/data_user_roles/data_cognito_user_roles.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    for var in ['role', 'user_pool_id', 'account_id', 'region']:
+        if var not in tf_input:
+            sys.stderr.write("Missing required variable {}".format(var))
+            sys.exit(1)
+
+    session = get_session(tf_input['role'], tf_input['region'])
+
+    users = get_cognito_users(session.client('cognito-idp'), tf_input['user_pool_id'])
+    users_with_roles = get_roles_for_users(map(lambda user_obj: user_obj['Username'], users), tf_input['account_id'])
```
Comment (IanUtley) [inline terraform/modules/data_user_roles/data_cognito_user_roles.py:45]: The Username you are using should not be 'Username', its preferred_name + first 3 chars of subject, or Username + first 3 chars of subject

File: terraform/modules/data_user_roles/data_cognito_user_roles.py (hunk 5 above)
Comment (IanUtley) [reply terraform/modules/data_user_roles/data_cognito_user_roles.py:25]: Change instead to just be active users.

File: terraform/deploy/app/modules.tf (hunk 3 above)
Comment (IanUtley) [reply terraform/deploy/app/modules.tf:5]: Hmm, I think this decides which account contains the ami, and looking at the provider, this is always management, possibly ok  to keep.

**Review by steveburton4 — APPROVED** [review]
